api/proto/teleport/healthcheckconfig/v1/health_check_config.proto

```diff
@@ -64,4 +64,11 @@ message Matcher {
   // empty value is ignored. The match result is logically ANDed with DBLabels,
   // if both are non-empty.
   string db_labels_expression = 2;
+  // KubernetesLabels matches Kubernetes labels. An empty value is ignored. The match
+  // result is logically ANDed with KubernetesLabelsExpression, if both are non-empty.
+  repeated teleport.label.v1.Label kubernetes_labels = 3;
+  // KubernetesLabelsExpression is a label predicate expression to match Kubernetes. An
+  // empty value is ignored. The match result is logically ANDed with KubernetesLabels,
+  // if both are non-empty.
+  string kubernetes_labels_expression = 4;
```

Comment:
What's the logic going to look like for an auth server or an agent that doesn't know about some of these fields? We don't currently have anything actually making use of `Matcher` as far as I can tell, but once we do we have to make sure we gracefully handle the case where the matcher appears to be "empty".